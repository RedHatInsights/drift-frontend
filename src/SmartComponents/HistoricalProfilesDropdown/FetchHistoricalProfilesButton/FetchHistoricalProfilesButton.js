import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Button } from '@patternfly/react-core';

class FetchHistoricalProfilesButton extends Component {
    constructor(props) {
        super(props);

        this.fetchHistoricalComparison = this.fetchHistoricalComparison.bind(this);
    }

    async fetchHistoricalComparison() {
        this.props.fetchCompare();
    }

    render() {
        return (
            <Button
                style={ { float: 'left' } }
                variant='primary'
                onClick={ this.fetchHistoricalComparison }>
                Compare
            </Button>
        );
    }
}

FetchHistoricalProfilesButton.propTypes = {
    fetchCompare: PropTypes.func
};

export default FetchHistoricalProfilesButton;
