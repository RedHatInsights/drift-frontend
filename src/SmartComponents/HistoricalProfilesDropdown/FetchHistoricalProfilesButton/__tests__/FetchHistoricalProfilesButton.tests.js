import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import FetchHistoricalProfilesButton from '../FetchHistoricalProfilesButton';

describe('FetchHistoricalProfilesButton', () => {
    it('should render correctly', () => {
        const wrapper = shallow(
            <FetchHistoricalProfilesButton />
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render mount correctly', () => {
        const wrapper = mount(
            <FetchHistoricalProfilesButton />
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should call fetchCompare', () => {
        const fetchCompare = jest.fn();

        const wrapper = mount(
            <FetchHistoricalProfilesButton
                fetchCompare={ fetchCompare }
            />
        );

        wrapper.find('.pf-c-button').simulate('click');
        expect(fetchCompare).toHaveBeenCalledTimes(1);
    });
});
