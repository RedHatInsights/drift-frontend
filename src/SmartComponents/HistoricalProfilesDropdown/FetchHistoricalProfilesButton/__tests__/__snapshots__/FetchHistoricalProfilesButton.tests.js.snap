// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FetchHistoricalProfilesButton should render correctly 1`] = `
<Component
  onClick={[Function]}
  style={
    Object {
      "float": "left",
    }
  }
  variant="primary"
>
  Compare
</Component>
`;

exports[`FetchHistoricalProfilesButton should render mount correctly 1`] = `
<FetchHistoricalProfilesButton>
  <Component
    onClick={[Function]}
    style={
      Object {
        "float": "left",
      }
    }
    variant="primary"
  >
    <ComponentWithOuia
      component={[Function]}
      componentProps={
        Object {
          "children": "Compare",
          "onClick": [Function],
          "style": Object {
            "float": "left",
          },
          "variant": "primary",
        }
      }
      consumerContext={null}
    >
      <Button
        onClick={[Function]}
        ouiaContext={
          Object {
            "isOuia": false,
            "ouiaId": null,
          }
        }
        style={
          Object {
            "float": "left",
          }
        }
        variant="primary"
      >
        <button
          aria-disabled={null}
          aria-label={null}
          className="pf-c-button pf-m-primary"
          disabled={false}
          onClick={[Function]}
          style={
            Object {
              "float": "left",
            }
          }
          tabIndex={null}
          type="button"
        >
          Compare
        </button>
      </Button>
    </ComponentWithOuia>
  </Component>
</FetchHistoricalProfilesButton>
`;
